apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/bytebuddy/ArgumentTypeNameMatch.java

```diff
@@ -60,7 +60,7 @@ private ArgumentTypeNameMatch(int index, String argumentTypeName) {
     public boolean matches(MethodDescription target) {
         ParameterList<?> parameters = target.getParameters();
         if (parameters.size() > index) {
-            return parameters.get(index).getType().asErasure().getName().equals(argumentTypeName);
```

Comment:
What does Array return in this case?

Reply:
When you read the Java class file spec, you would notice `[Lorg.apache.skywalking.apm.agent.core.plugin.bytebuddy.Person` is actually the correct expression for `Person[]`

You could learn this from Java specification. This is official.

Reply:
Yes, I know that.

But parameters.get(index).getType().asErasure().getName() will cause some Instrumentation classes fail, e.g. org.apache.skywalking.apm.plugin.jdbc.postgresql.define.ConnectionInstrumentation, and org.apache.skywalking.apm.plugin.jdbc.postgresql.define.PgStatementInstrumentation.

Of course,  we can let plugin authors write in "[Lorg.apache.skywalking.apm.agent.core.plugin.bytebuddy.Person". But don't you think "Person[]" is better for readability?

Reply:
> But parameters.get(index).getType().asErasure().getName() will cause some Instrumentation classes fail, e.g. org.apache.skywalking.apm.plugin.jdbc.postgresql.define.ConnectionInstrumentation, and org.apache.skywalking.apm.plugin.jdbc.postgresql.define.PgStatementInstrumentation.

What does this `but` mean? They are working before your change.

Reply:
No,they cannot define the InterceptPoint correctly. 
  Because java.lang.String[]  is not equal to [java.lang.String in org.apache.skywalking.apm.plugin.jdbc.postgresql.define.ConnectionInstrumentation

Reply:
OK, maybe tests miss those cases, so don't detect it. 

I think you should fix those rather than changing this API implementation. Also, I would recommend to updating plugin dev doc to highlight this.

Reply:
java.lang.String[] is more friendly and lower knowledge requirement for plugin authors than [java.lang.String, isn't it?

Reply:
> java.lang.String[] is more friendly and lower knowledge requirement for plugin authors than [java.lang.String, isn't it?

This is a kind of instrumentation, so, we should be closing to Java class file format, rather than Java source codes. 
Let's fix those wrong instrumentation definitions, to use `[L`. Also, we should add a parameter check for `ArgumentTypeNameMatch`. If the given value is using `[]` suffix, the kernel should report an exception with a clear message saying, please use `[L` to define an Array type, and ref to Java specification.

Would you like to open another pull request to fix these?

Reply:
Another similar story in the backend OAL engine, as it is on the top of Javassist, another binary code manipulator, we can't use `int-Integer` boxing and unboxing, as it doesn't exist in the binary code level at all.
So, we don't want to hide this tech details, we could add verifications to avoid the bugs you mentioned above.